### PR TITLE
Chore/update all packages

### PR DESCRIPTION
**Pull request by olastor**
Updates many packages. `package.json` doesn't seem to show the changes, but `yarn.lock` does???

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,29 +57,29 @@
...
     "karma-jasmine-html-reporter": "^0.2.2",
-    "ng2-mock-component": "^0.0.6",
+    "ng2-mock-component": "^0.1.0",
     "protractor": "^5.1.2",
     "sw-precache": "^5.2.0",
-    "ts-node": "~3.2.0",
+    "ts-node": "~3.3.0",
     "tslint": "^5.5.0",
-    "typescript": "~2.4.1"
+    "typescript": "^2.3.4"
```
Comment (olastor) [inline package.json]: This was downgraded because of angular cli warning. Valid?

Comment (binarious) [reply package.json]: yep

Comment (binarious) [reply package.json]: https://github.com/angular/angular-cli/issues/7625

**Comment by binarious** [comment]
Switched to https://github.com/angelozerr/tslint-language-service/blob/master/README.md#visual-studio-code in order to get errors about unused imports/variables. Check the page to get information about text editor support.

**Comment by binarious** [comment]
Reduced app size from 8.07 MB to 3.78 MB

**Pull request merged**